### PR TITLE
Remove DIR_DELIM in builtin Lua

### PR DESCRIPTION
**Pull request by rubenwardy**
Lua is a cross-platform language. The standard library allows you to use forward slash on Windows, so that Lua code doesn't need to care about whether the platform uses a different path separator.

This PR makes builtin Lua use "/" instead of "DIR_DELIM. This improves the code quality and readability of parts by a fair amount.

The value DIR_DELIM has not been changed, however it should be deprecated

## To do

This PR is Ready for Review

### CODE REVIEW

File: builtin/mainmenu/pkgmgr.lua [hunk 1]
```diff
@@ -17,15 +17,15 @@
 
 --------------------------------------------------------------------------------
 local function get_last_folder(text,count)
-	local parts = text:split(DIR_DELIM)
+	local parts = text:split("/")
```
Comment (rubenwardy) [inline builtin/mainmenu/pkgmgr.lua:20]: This only works if the engine always provides paths with forward slashes. I've attempted to do this in l_mainmenu, but another option would be

```lua
text:gsub("[/\\]+", "/"):split("/")
```

**Comment by sfan5** [comment]
> This PR makes DIR_DELIM always equal "/" in Lua

This would break mods that use `DIR_DELIM == "\\"` as a platform check, not sure how common this idiom is.

**Comment by rubenwardy** [comment]
> This would break mods that use DIR_DELIM == "\\" as a platform check, no idea how common this idiom is

DIR_DELIM is undocumented so this is fine from a versioning perspective. In practice, I couldn't find any mods on ContentDB that do this ( https://content.minetest.net/zipgrep/23a149b0-393a-4b69-82d9-a584e801a3a5/ )

The bigger problem would be if mods are using third-party libraries that return native paths and are relying on DIR_DELIM to have the right value to `:split("/")`.

**Comment by fluxionary** [comment]
i honestly don't have any real motivation to support my mods outside of posix environments, but i make use of DIR_DELIM extensively just as a "best practice". it *sounds* like lua will have my back when that assumption fails? there's certainly edge cases that will fail though, no? particularly for paths that involve backslashes (windows) and colons (windows, classic mac os (though does lua even run on classic mac os (and who even runs classic mac os (and i don't care about classic mac os))))

**Comment by rubenwardy** [comment]
> i honestly don't have any real motivation to support my mods outside of posix environments, but i make use of DIR_DELIM extensively just as a "best practice". it _sounds_ like lua will have my back when that assumption fails? there's certainly edge cases that will fail though, no? particularly for paths that involve backslashes (windows) and colons (windows, classic mac os (though does lua even run on classic mac os (and who even runs classic mac os (and i don't care about classic mac os))))

You can already just use `/` for dofile and building paths, in fact I'd recommend doing this for your own sanity. The only issues would be when parsing paths (you should accept both slashes) or when passing paths to a Minetest API that doesn't do paths in a portable way (I don't know of any, this would be any engine)

TL;DR: feel free to use use `/` when building paths, accept both `\` and `/` when reading paths

**Comment by appgurueu** [comment]
modlib uses `DIR_DELIM`, but it should gracefully fall back to `/` if it is absent:

```lua
local dir_delim = rawget(_G, "DIR_DELIM") -- Minetest
	or (rawget(_G, "package") and package.config and assert(package.config:match("^(.-)[\r\n]"))) -- non-Minetest
	or "/"
```

Small UX side note: I'm pretty sure Minetest does no postprocessing when showing paths to Windows users, so they will often be shown paths (e.g. in stacktraces, IIRC) like `C:\blah\blub/here/is/where/the/modder/concatted`. Ideally these should be normalized before display (might require overriding `debug.traceback`, probably not worth it).

**Comment by rubenwardy** [comment]
I've removed the `DIR_DELIM = "/"` change from this PR as I feel like it would be too disruptive. All that remains is using "/" in builtin. Thoughts welcome on what we should here / whether this is worth doing

**Comment by rubenwardy** [comment]
closing as no interest
